lib/model/folder.go

```diff
@@ -78,9 +79,10 @@ func newFolder(model *Model, cfg config.FolderConfiguration) folder {
 
 		pullScheduled: make(chan struct{}, 1), // This needs to be 1-buffered so that we queue a pull if we're busy when it comes.
 
-		watchCancel: func() {},
-		watchErr:    errWatchNotStarted,
-		watchErrMut: sync.NewMutex(),
+		restartWatchChan: make(chan struct{}, 1),
```

Comment:
Where is this channel used?

Reply:
https://github.com/imsodin/syncthing/blob/aa470b5303f6774cd4eaad7d0f71efb51c4f97d2/lib/model/folder.go#L166

This is like *the* main fix of this PR, the rest is UI cosmetics :)

Reply:
It's not used in this pr tho, so I am puzzled.

Reply:
The problem was, that before `restartWatchChan` never was initialized, so always was `nil` -> the watch never restarted after a failure at any point, even though everything was in place already.